planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/test/test_utils.cpp

```diff
@@ -12,46 +12,1540 @@
+    ObjectData object;
+    object.direction = Direction::RIGHT;
+
+    ASSERT_TRUE(isSameDirectionShift(isOnRight(object), negative_shift_length));
+    ASSERT_FALSE(isSameDirectionShift(isOnRight(object), positive_shift_length));
+  }
+
+  {
+    ObjectData object;
+    object.direction = Direction::LEFT;
+    ASSERT_TRUE(isSameDirectionShift(isOnRight(object), positive_shift_length));
+    ASSERT_FALSE(isSameDirectionShift(isOnRight(object), negative_shift_length));
+  }
+
+  {
+    ObjectData object;
+    object.direction = Direction::NONE;
+    EXPECT_THROW(isSameDirectionShift(isOnRight(object), positive_shift_length), std::logic_error);
+  }
+}
+
+TEST(TestUtils, isShiftNecessary)
+{
+  constexpr double negative_shift_length = -1.0;
+  constexpr double positive_shift_length = 1.0;
+
+  {
+    ObjectData object;
+    object.direction = Direction::RIGHT;
+
+    ASSERT_TRUE(isShiftNecessary(isOnRight(object), positive_shift_length));
+    ASSERT_FALSE(isShiftNecessary(isOnRight(object), negative_shift_length));
+  }
 
-TEST(BehaviorPathPlanningAvoidanceUtilsTest, shiftLengthDirectionTest)
+  {
+    ObjectData object;
+    object.direction = Direction::LEFT;
+    ASSERT_TRUE(isShiftNecessary(isOnRight(object), negative_shift_length));
+    ASSERT_FALSE(isShiftNecessary(isOnRight(object), positive_shift_length));
+  }
+
+  {
+    ObjectData object;
+    object.direction = Direction::NONE;
+    EXPECT_THROW(isShiftNecessary(isOnRight(object), positive_shift_length), std::logic_error);
+  }
+}
+
+TEST(TestUtils, calcShiftLength)
+{
+  {
+    constexpr bool is_on_right = true;
+    constexpr double overhang = -1.0;
+    constexpr double margin = 1.5;
+
+    const auto output = calcShiftLength(is_on_right, overhang, margin);
+    EXPECT_DOUBLE_EQ(output, 0.5);
+  }
+
+  {
+    constexpr bool is_on_right = false;
+    constexpr double overhang = -1.0;
+    constexpr double margin = 1.5;
+
+    const auto output = calcShiftLength(is_on_right, overhang, margin);
+    EXPECT_DOUBLE_EQ(output, -2.5);
+  }
+}
+
+TEST(TestUtils, insertDecelPoint)
+{
+  // invalid target decel point
+  {
+    auto path = create_test_path();
+
+    const auto ego_position = createPoint(2.5, 0.5, 0.0);
+    constexpr double offset = 100.0;
+    constexpr double velocity = 1.0;
+
+    std::optional<geometry_msgs::msg::Pose> p_out{std::nullopt};
+    insertDecelPoint(ego_position, offset, velocity, path, p_out);
+
+    EXPECT_FALSE(p_out.has_value());
+    std::for_each(path.points.begin(), path.points.end(), [](const auto & p) {
+      EXPECT_DOUBLE_EQ(p.point.longitudinal_velocity_mps, 20.0);
+    });
+  }
+
+  // nominal case
+  {
+    auto path = create_test_path();
+
+    const auto ego_position = createPoint(3.5, 0.5, 0.0);
+    constexpr double offset = 3.0;
+    constexpr double velocity = 1.0;
+
+    std::optional<geometry_msgs::msg::Pose> p_out{std::nullopt};
+    insertDecelPoint(ego_position, offset, velocity, path, p_out);
+
+    EXPECT_TRUE(p_out.has_value());
```

Comment:
If p_out.has_value() is FALSE, L860~862 would have segmentation fault. So just in case.
```suggestion
   ASSERT_TRUE(p_out.has_value());
```

Reply:
fix in https://github.com/autowarefoundation/autoware.universe/pull/9134/commits/34b64c36a5fa9a1a4c9972e355f1f9ce4b7c0cd6.